exporters/otlp/internal/transform/span.go

```diff
@@ -32,14 +33,14 @@ func SpanData(sdl []*export.SpanData) []*tracepb.ResourceSpans {
 	if len(sdl) == 0 {
 		return nil
 	}
-	// Group by the unique string representation of the Resource.
-	rsm := make(map[string]*tracepb.ResourceSpans)
+	// Group by the distinct representation of the Resource.
+	rsm := make(map[label.Distinct]*tracepb.ResourceSpans)
 
 	for _, sd := range sdl {
 		if sd != nil {
-			var key string
+			var key label.Distinct
 			if sd.Resource != nil {
```

Comment:
These guards aren't necessary now that `Resrource.Equivalent()` handles `nil` correctly.

Reply:
Done.